### ISSUE TITLE
I don't want to go back to last cursor position when navigating back in C/C++

### ISSUE THREAD

**Issue by pharsh22** [microsoft/vscode#146875]
Hi team,

Here's the scenario:

void foo_bar()
{
    ...
    ...
    this->myFunction();   => go to definition at myFunction
}

void myFunction()
{
   ... a bunch of lines...
   For( : : )
   {
      // after coming to this function, I press Page Down and get to this line
   }

   this->someOtherFunction
}

Sequence of events:
1. Hit F12 on `this->myFunction`, reach the first line of the function definition
2. Hit pagedown inside `myFunction`, reach some line x. Inside `myFunction`, hit F12 on `this->someOtherFunction` and the cursor is now at the first line of `someOtherFunction`.
3. Hit Alt + <-  : this brings me back to `this->someOtherFunction` line inside `myFunction`. Hit Alt + <- again and this brings me to line x. I don't want to go to line x. I want to go back to line `this->myFunction` in `foo_bar`.

I don't want to record every cursor location in my navigation. I just want to go from one API to another (NOT within every single cursor stop in my navigation)

Is there a way to do this? If not, can I please request a feature for this?

Sorry if this is a duplicate, after extensive research, I'm not able to find a solution to this. I found a bunch of topics wanting to record last cursor position in navigation which is exactly opposite of what I want to do.

Thanks,
Harsh

**Comment by bpasero** [comment]
Does https://code.visualstudio.com/updates/v1_65#_new-editor-history-navigation help?

**Comment by pharsh22** [comment]
Fantastic! I did stumble upon it but didn't read carefully I guess (my bad!)

I have now added key bindings for these:
workbench.action.navigateForwardInNavigationLocations - Go forward in navigation locations.
workbench.action.navigateBackInNavigationLocations

And can navigate accordingly. (I also checked internal tools but couldn't find anything reliable. I'll ping you internally if I have further questions)

Thanks Ben!
